libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -453,10 +453,19 @@ bool ASTJsonConverter::visit(ModifierDefinition const& _node)
 
 bool ASTJsonConverter::visit(ModifierInvocation const& _node)
 {
-	setJsonNode(_node, "ModifierInvocation", {
+	std::vector<pair<string, Json::Value>> attributes{
 		make_pair("modifierName", toJson(_node.name())),
 		make_pair("arguments", _node.arguments() ? toJson(*_node.arguments()) : Json::nullValue)
-	});
+	};
+	Declaration const* declaration = _node.name().annotation().referencedDeclaration;
+	if (declaration)
+	{
+		if (dynamic_cast<ModifierDefinition const*>(declaration))
+			attributes.emplace_back("kind", "modifierInvocation");
+		else if (dynamic_cast<FunctionDefinition const*>(declaration))
+			attributes.emplace_back("kind", "baseConstructorSpecifier");
```

Comment:
Can we add an assert about `isConstructor()` here?

Reply:
I'm not sure this works with "parseOnly".

Reply:
I'll leave out "kind" if it is not a constructor.